OpenRA.Mods.Common/Traits/Conditions/GrantCondition.cs

```diff
@@ -50,7 +53,8 @@ protected override void TraitDisabled(Actor self)
 			if (conditionToken == ConditionManager.InvalidConditionToken)
 				return;
 
-			conditionToken = conditionManager.RevokeCondition(self, conditionToken);
+			if (!Info.GrantPermanently)
```

Comment:
Imho it would make sense to add this to the if statement above, so the code would look like this:
```
if (Info.GrantPermanently || conditionToken == ConditionManager.InvalidConditionToken)
	return;

conditionToken = conditionManager.RevokeCondition(self, conditionToken);
```

Reply:
Thanks.